pkg/cfn/builder/templates/beta.py

```diff
@@ -0,0 +1,405 @@
+
+        # Validate input
+        validate_input(event)
+
+        iam_principal_arn = event['ResourceProperties']['IAMPrincipalArn']
+        sts_role_arn = event['ResourceProperties']['STSRoleArn']
+
+        # Prepare the create_cluster request payload from the custom resource properties
+        create_cluster_payload = convert_keys_to_lowercase_first_letter(
+            copy.deepcopy(event['ResourceProperties']))
+
+        # cleanup create_cluster_payload as not all fields can be sent to EKS create_cluster API
+        del create_cluster_payload['serviceToken']
+        del create_cluster_payload['iAMPrincipalArn']
+        del create_cluster_payload['sTSRoleArn']
+        replace_boolean_strings(create_cluster_payload)
+
+        update_payload_tags(create_cluster_payload, event)
+
+        # create and wait for the eks cluster
+        cluster_details, response = create_cluster(eks_client, cluster_name, create_cluster_payload)
+
+        # Create an access entry for the EKS cluster
+        create_access_entry(eks_client, iam_principal_arn, sts_role_arn, cluster_name, "STANDARD")
+
+        # Extract required attributes
+        eventData = {
+            "Arn": cluster_details['arn'],
+            "PhysicalResourceId": cluster_details['arn'],
+            "ClusterName": cluster_details['name'],
+            "ClusterSecurityGroupId": cluster_details['resourcesVpcConfig']['clusterSecurityGroupId'],
+            "CertificateAuthorityData": cluster_details['certificateAuthority']['data'],
+            "Endpoint": cluster_details['endpoint'],
+        }
+
+        cfnresponse.send(event, context, cfnresponse.SUCCESS,
+                         eventData)
+        # Return the cluster ARN as the PhysicalResourceId
+        return {
+            'PhysicalResourceId': response['cluster']['arn'],
+            "ClusterName": response['cluster']['name'],
+            'Data': json.dumps(response, default=str)
+        }
+    except Exception as e:
+        logger.error("Error: " + str(e))
+        cfnresponse.send(event, context, cfnresponse.FAILED, {"Message": str(e)})
+        return None
+
+
+def create_cluster(eks_client, cluster_name, create_cluster_payload):
+    """
+    Create the EKS cluster
+    """
+    try:
+        # Check if the cluster already exists
+        logger.info(f"Checking if EKS cluster {cluster_name} already exists")
+        response = eks_client.describe_cluster(name=cluster_name)
+    except eks_client.exceptions.ResourceNotFoundException:
+        logger.info("Creating EKS cluster with payload: " + json.dumps(create_cluster_payload, default=str))
+        response = eks_client.create_cluster(**create_cluster_payload)
+        logger.info("EKS cluster created: " + json.dumps(response, default=str))
+
+    # Wait for the cluster to become ACTIVE
+    cluster_details = wait_for_cluster_creation(eks_client, cluster_name)
+    return cluster_details, response
+
+
+def wait_for_cluster_creation(eks_client, cluster_name):
```

Comment:
nit: boto3 has built-in waiters that you can use so you don't have to re-implement your own - https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/eks/waiter/ClusterActive.html

Reply:
cool, will revisit

Reply:
Done!